### PR TITLE
tickets/PREOPS-5118: Update the schedview snapshot dashboard running in phalanx to support access to an S3 bucket

### PR DESCRIPTION
**Pull request by ehneilsen**
This change requires not just updating the schedview version, but also requires configuring phalanx to provide it with the secret it needs.
For this, I was guided by a combination of the phalanx documentation, the configuration of plot-navigator (from which some changes were cut-and-pasted), and [this thread on slack](https://lsstc.slack.com/archives/C2JP8GGVC/p1714426470147829).
It seems to work like this, but I'm not 100% sure copying the secret from nublado using the butler's vault is the right thing to do, in that I think it give this app write access it doesn't need (although KT [says this is okay](https://lsstc.slack.com/archives/C2JP8GGVC/p1714580872053049?thread_ts=1714426470.147829&cid=C2JP8GGVC)).

### CODE REVIEW

File: applications/schedview-snapshot/secrets.yaml
Comment (rra) [inline applications/schedview-snapshot/secrets.yaml:9]: Does this application need all of the credentials that a user would have, to all of the S3 buckets, or does it only need access to a specific subset of S3 buckets? I'm worried that this is too many permissions and it would be better to create more scoped credentials specific to the application, but I'm not sure what the application does. If it should be able to access anything a user can access from a notebook (read and write), then this is correct.

File: applications/schedview-snapshot/templates/_helpers.tpl
Comment (rra) [inline applications/schedview-snapshot/templates/_helpers.tpl:25]: We're no longer using this pattern with any new applications because it's unnecessary complexity that's never used in Phalanx. I recommend deleting this all again and just putting schedview-snapshot literally in your YAML files.

File: applications/schedview-snapshot/templates/deployment.yaml
Comment (rra) [inline applications/schedview-snapshot/templates/deployment.yaml:87]: Here, for example, this should just be "schedview-snapshot" instead of this more complex include template.

File: applications/schedview-snapshot/templates/vault-secrets.yaml
Comment (rra) [inline applications/schedview-snapshot/templates/vault-secrets.yaml:10]: The butler-secret pattern is no longer supported and we're trying to move away from it. You'll see that plot-navigator has an if conditional on separateSecrets. Since this is a new secret, you should use only the separateSecrets approach, which is the new way to do this.

In other words, path should instead be:
```
path: "{{ .Vault.global.vaultSecretsPath }}/schedview-snapshot"
```

File: applications/schedview-snapshot/templates/vault-secrets.yaml
Comment (rra) [inline applications/schedview-snapshot/templates/vault-secrets.yaml:6]: Here too, change this to just "schedview-snapshot"

**Review by rra — COMMENTED** [review]
Note that only usdfint has switched to the new secrets management system, so for usdfdev you (or one of the USDF administrators) will need to manually put the AWS credentials that your application needs into the aws-credentials.ini key under the schedview-snapshot Vault path for your application.

File: applications/schedview-snapshot/secrets.yaml
Comment (ehneilsen) [reply applications/schedview-snapshot/secrets.yaml:9]: The application needs read access to the LFA (is this just one bucket, or many?): all of URIs for objects I need access to begin `s3://rubin:rubinobs-lfa-cp/Scheduler`. (There will need to be a deployment at the summit as well, with access to the corresponding summit URIs, but I'm not there yet, and summit staff will probably be the ones to configure phalanx for that environment, and it will be handled in a separate ticket and pull request.)
I don't know how to get credentials that limit the access to this subset. When I asked on slack (see the thread I linked to in the original comment), it sounded like trying this would make things more complicated, and it would be acceptable and simpler to use the nublado credentials. If I can get help getting something more restricted, that would be great.

File: applications/schedview-snapshot/templates/vault-secrets.yaml
Comment (ehneilsen) [reply applications/schedview-snapshot/templates/vault-secrets.yaml:10]: How do I create (or who do I ask to create) an entry in the vault for schedview-snapshot? How do I create (or who do I ask to create) the secrets that need to go into the vault?

File: applications/schedview-snapshot/templates/_helpers.tpl
Comment (ehneilsen) [reply applications/schedview-snapshot/templates/_helpers.tpl]: Change made in commit ca02a336.

File: applications/schedview-snapshot/templates/deployment.yaml
Comment (ehneilsen) [reply applications/schedview-snapshot/templates/deployment.yaml]: Change made in commit ca02a336.

File: applications/schedview-snapshot/secrets.yaml
Comment (rra) [reply applications/schedview-snapshot/secrets.yaml:9]: Well, my security advice would be to get a more restricted token that only has read access to the objects or bucket that you are using, put that token in the value for the `aws-credentials.ini` secret in the Vault path for this application, and remove the copy directive (eveything else about this `secrets.yaml` is file). Using correctly-scoped tokens is generally better, in part because it will protect against mistakes.

But I'm not involved in running the USDF cluster and it's their security decision, so if they really want you to use the more-powerful tokens that are injected into user notebooks, it's their call. @ktlim, are you the person who handles this?

File: applications/schedview-snapshot/templates/vault-secrets.yaml
Comment (rra) [reply applications/schedview-snapshot/templates/vault-secrets.yaml:10]: You can ask on #ops-usdf on Slack for help with Vault. I'm not sure if they'll want to give you access so that you can make the change yourself or make it for you. Where you get the secret depends on the answer to the above question: either someone should give you a new secret, or you (or someone helping you) will get it from the nublado Vault secret.

File: applications/schedview-snapshot/secrets.yaml
Comment (ktlim) [reply applications/schedview-snapshot/secrets.yaml:9]: The rubin-summit-users credential is currently used for read-only access to all buckets that staff users will need access to on an everyday basis so that multiple profiles don't need to be used.  (The credential also has read/write access to the buckets containing Butler repo user collections.)

In the near future (within the next few weeks) when we move to the embargo rack, which necessitates multiple profiles anyway due to having multiple endpoint URLs, I have been considering having an additional LFA-specific read-only credential that would be used with `s3://lfa@rubinobs-lfa-cp/Scheduler` URIs.  That profile would also be installed in the nublado `butler-secret`, but a more limited custom LFA-only profile could also be used.

Comment (rra) [reply applications/schedview-snapshot/secrets.yaml:9]: It sounds like it'll be a couple of weeks before things are ready to switch to more limited secrets, so this copy directive is correct for now. But you'll still need to change the reference to butler-secret to schedview-snapshot instead, down below where you create the `VaultSecret` resource, since we're switching everything over to per-app secrets. usdfint is done, and usdfdev and maybe even usdfprod should hopefully be done next Thursday.

Once the more restrictive secrets have been implemented, the more restricted secret should be the only one included in the schedview-snapshot secret, and at that point the copy directive here should be removed.

**Review by rra — APPROVED** [review]

**Pull request merged**
